etl-destinations/src/iceberg/encoding.rs

```diff
@@ -0,0 +1,1288 @@
+use std::sync::Arc;
+
+use arrow::{
+    array::{
+        ArrayRef, ArrowPrimitiveType, BooleanBuilder, FixedSizeBinaryBuilder, LargeBinaryBuilder,
+        PrimitiveBuilder, RecordBatch, StringBuilder, TimestampMicrosecondBuilder,
+    },
+    datatypes::{
+        DataType, Date32Type, Float32Type, Float64Type, Int32Type, Int64Type, Schema,
+        Time64MicrosecondType, TimeUnit, TimestampMicrosecondType,
+    },
+    error::ArrowError,
+};
+// use base64::{Engine, prelude::BASE64_STANDARD};
+use chrono::{NaiveDate, NaiveTime};
+use etl::types::{Cell, TableRow};
+
+pub const UNIX_EPOCH: NaiveDate =
+    NaiveDate::from_ymd_opt(1970, 1, 1).expect("unix epoch is a valid date");
+
+const MIDNIGHT: NaiveTime = NaiveTime::from_hms_opt(0, 0, 0).expect("midnight is a valid time");
+
+const UUID_BYTE_WIDTH: i32 = 16;
+
+/// Converts a slice of [`TableRow`]s to an Arrow [`RecordBatch`].
+///
+/// This function transforms tabular data from the ETL pipeline's internal format
+/// into Apache Arrow's columnar format for efficient storage and processing in
+/// Iceberg tables. Each field in the schema is processed sequentially to build
+/// the corresponding Arrow arrays.
+///
+/// # Arguments
+///
+/// * `rows` - A slice of [`TableRow`] instances containing the data to convert
+/// * `schema` - The Arrow [`Schema`] defining the structure and types of the output batch
```

Comment:
nit: we generally don't use the arguments section in docs, if not strictly needed I would keep rid of it. It adds more things that we have to maintain in the docs.